### DISCUSSION TITLE
Is it possible to output both Json and Junit

### DISCUSSION THREAD

**Discussion by ecaponi3** [cucumber-rs/cucumber#287] [Q&A] [answered]
Looking into ways to use the built in writer's to produce both a json test report as well as a junit xml report from the same run.

Here is an example of how we are using .tee to produce our json file currently: 

`let world =
world.with_writer(cucumber::writer::Basic::stdout()
.summarized()
.tee(cucumber::writer::Json::new(driver.raw_json_file()?)),
);`
`world.run(products_dir).await; `
        

Is it possible to call on tee again to produce the same report but in xml? Or is there another built in method that would be useful for this?

**Comment by ilslv** [comment, marked as answer]
@ecaponi3  yes, you can use `WriterExt::tee()` to chain any number of `Writer`s:

```rust
.with_writer(
    writer::Basic::stdout()
        .summarized()
        .tee(writer::Json::new(json.reopen().unwrap()))
        .tee(writer::JUnit::new(junit.reopen().unwrap(), 1)),
)
```

You can also speed up computations a bit by using non-normalized `Writers` in `WriterExt::tee()` and lift a single normalization:

```rust
.with_writer(
    writer::Basic::raw(io::stdout(), Coloring::Auto, 0)
        .summarized()
        .tee(writer::Json::for_tee(json))
        .tee(writer::JUnit::for_tee(junit, 1))
        .normalized(),
)
```

**Reply by tyranron** [reply]
@ecaponi3 this topic is also covered in [5.4. Multiple outputs](https://cucumber-rs.github.io/cucumber/current/output/multiple.html) Book chapter.

**Reply by ecaponi3** [reply]
Awesome thank you both this is super helpful!
